CompilerSource/parsing/lexer.cpp

```diff
@@ -422,7 +422,7 @@ bool Lexer::HandleMacro(std::string_view name) {
         else if (t.type == TT_ENDPARENTH) --paren;
         if (t.type == TT_COMMA && paren == 1) {
           args.emplace_back();
-        } else {
+        } else if (paren != 0) {
```

Comment:
This is where the variadic check should be done, too... I just forgot. 😜

Reply:
Well, I feel variadic-ness should be handled in `SubstituteAndUnroll` because this function just seems to handle reading in n-number of arguments but not validating any of them.

Reply:
But then we lose the comma tokens that separated these variadic arguments. To keep the original tokens, we need to have that context here.

Reply:
I don't think that matters? As the macro arguments are stored as a `std::vector<TokenVector>`, each element of the vector contains all the tokens making up each argument, so it doesn't matter if commas are stored or not.

Reply:
We can always spoof a new comma Token when we need to start returning the variadic argument tokens, sure, but that's extra processing we need to do on the spot. Usually, we never see the comma token again, but if your variadic args are "a, b", `__VA_ARGS__` needs to include that comma (and all the others). We lose the source information this way (e.g. the original code pointer), so if that comma turns out to add too many arguments to an underlying normal function, for example, the error reporting might get very confusing.